js/libs/form.js

```diff
@@ -114,31 +114,67 @@ Fliplet.Widget.instance('form-builder', function(data) {
 
     if (fields.length && (data.saveProgress && typeof progress === 'object') || entry) {
       fields.forEach(function(field) {
-        if (entry && entry.data && typeof entry.data[field.name] !== 'undefined' && field.populateOnUpdate !== false) {
-          if (field._type === "flDate") {
-            if (Fliplet.Env.get('platform') === 'web') {
-              field.value = moment(entry.data[field.name]).format('DD MMMM YYYY');
-            } else {
-              field.value = moment(entry.data[field.name]).format('YYYY-MM-DD');
-            }
-          } else if (field._type === "flCheckbox" && !Array.isArray(entry.data[field.name])) {
-            field.value = [];
-          } else if (field._type === 'flImage') {
-            var img = entry.data[field.name];
-            field.value = [];
-            if (Array.isArray(img)) {
-              field.value = img;
-            } else if (typeof img === 'string') {
-              field.value.push(img);
-            }
+        if (entry && entry.data && field.populateOnUpdate !== false) {
+          switch (field._type) {
+            case 'flDate':
+              var regex = /([12]\d{3}-(0[1-9]|1[0-2])-(0[1-9]|[12]\d|3[01]))/;
+              field.value = regex.exec(entry.data[field.name]) ? entry.data[field.name] : moment().get().format('YYYY-MM-DD');
+              break;
+
+            case 'flImage':
+            case 'flFile':
+              var img = entry.data[field.name];
+              field.value = [];
+
+              if (Array.isArray(img)) {
+                field.value = img;
+              } else if (typeof img === 'string') {
+                field.value.push(img);
+              }
+    
+              field.value = field.value.map(function(url) {
+                return Fliplet.Media.authenticate(url);
+              });
+              break;
 
-            field.value = field.value.map(function (url) {
-              return Fliplet.Media.authenticate(url);
-            });
-          } else if (!field._submit && typeof field._submit !== 'undefined') {
-            field.value = field.value;
-          } else {
-            field.value = entry.data[field.name];
+            case (!field._submit && typeof field._submit !== 'undefined' || typeof entry.data[field.name] === 'undefined'):
+              field.value = field.value;
```

Comment:
this feels useless, can you explain?

Reply:
@squallstar I wasn't adding this. It already was in the file. As far as I understand this line saves field.value in the case when the submit button isn't available and when new data is `undefined`.

Reply:
@YaroslavOvdii the original code is `!field._submit && typeof field._submit !== 'undefined'`. Your change breaks it because you have `a && b || c`. Have you tested this?

Reply:
@squallstar Yes, I have tested it. It worked.
The reason why I was adding or condition here is because of in this case `typeof entry.data[field.name] === 'undefined'` we were skipping all other if condition and just returning `field.value` which is equal for this condition as I can see. 
Was it used for another purpose?